src/directwrite/mod.rs

```diff
@@ -132,14 +143,61 @@ impl DirectWriteRasterizer {
 
         fallback_result.mapped_font
     }
+
+    /// Given a buffer containing a rasterized glyph, return a buffer with
+    /// three bytes per pixel regardless of the DirectWrite texture type in
+    /// use.
+    fn normalize_buffer(&self, buffer: Vec<u8>) -> Vec<u8> {
+        match self.texture_type {
+            dwrote::DWRITE_TEXTURE_CLEARTYPE_3x1 => buffer,
+            dwrote::DWRITE_TEXTURE_ALIASED_1x1 => {
+                let mut norm_buf: Vec<u8> = Vec::with_capacity(buffer.len() * 3);
+                for pix in buffer.iter() {
+                    norm_buf.push(*pix);
+                    norm_buf.push(*pix);
+                    norm_buf.push(*pix);
+                }
+                norm_buf
+            },
+            _ => panic!("invalid texture type"),
+        }
+    }
 }
 
 impl crate::Rasterize for DirectWriteRasterizer {
     fn new(device_pixel_ratio: f32, _: bool) -> Result<DirectWriteRasterizer, Error> {
+        let mut use_smoothing: BOOL = 0;
+        unsafe {
+            SystemParametersInfoA(
+                SPI_GETFONTSMOOTHING,
+                0,
+                &mut use_smoothing as *mut _ as *mut c_void,
+                0,
+            );
+        }
+
+        let mut smoothing_type: UINT = FE_FONTSMOOTHINGSTANDARD;
+        if use_smoothing != 0 {
+            unsafe {
+                SystemParametersInfoA(
+                    SPI_GETFONTSMOOTHINGTYPE,
+                    0,
+                    &mut smoothing_type as *mut _ as *mut c_void,
+                    0,
+                );
+            }
+        }
```

Comment:
Why do we have to check twice here if font smoothing should be enabled or not? That seems rather strange.